From 8931305205d20e6e261b72f5f699b08234f430eb Mon Sep 17 00:00:00 2001
From: Yoni Izutskiver <yizutskiver@habana.ai>
Date: Thu, 04 Nov 2021 20:09:01 +0200
Subject: [PATCH] [SW-0] ppboot: gaudi2: fix pcie phy init config

the default rtl values for protocol0_ext_rx_adapt_sel_g4 and
protocol0_ext_rx_adapt_mode_g3 are not correct. Initialized them
with 0xFF and 0x5555

Also corrected srs mode enable condition check

Change-Id: Ia34cc9204f98ff35d04c3f0e38862c850d2b5ea5
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index e65fc3d..5c7e385 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -399,7 +399,12 @@
 }
 
 static void pcie_phy_set_ext_tx_rx_ctrl(void) {}
-static void pcie_phy_set_protocol0_ext_tx_rx(void) {}
+static void pcie_phy_set_protocol0_ext_tx_rx(void) {
+	regmap_writel(PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_MODE_G3_CONF,
+			mmPCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_MODE_G3);
+	regmap_writel(PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_SEL_G4_CONF,
+			mmPCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_SEL_G4);
+}
 
 static uint8_t get_srns_mode_from_strap(uint32_t boot_strap)
 {
@@ -612,7 +617,7 @@
 	/* SRNS mode enable */
 	srns = get_srns_mode_from_strap(boot_strap);
 
-	if (!srns)
+	if (srns)
 		separate_reference = 1;
 
 	/* spread spectum clock enable */
